lib/manageiq/messaging/kafka/common.rb

```diff
@@ -0,0 +1,77 @@
+module ManageIQ
+  module Messaging
+    module Kafka
+      module Common
+        private
+
+        def producer
+          @producer ||= kafka_client.producer
+        end
+
+        def consumer
+          unless @consumer
+            raise "group_ref is missing from client initialization" unless @group_ref
+            @consumer = kafka_client.consumer(:group_id => @group_ref)
+          end
+          @consumer
+        end
+
+        trap("TERM") do
+          @consumer.stop if @consumer
```

Comment:
Could we provider an API on `ManageIQ::Messaging::Client` for this like `client.stop` or `client.terminate`?

Reply:
Actually that didn't work, I had to add a new method which called `@consumer.stop`.

Reply:
I will then close the consumer in the `client.close` method

Reply:
:+1: perfect